js/libs/UserMedia.js

```diff
@@ -0,0 +1,88 @@
+class UserMedia {
+  static async getStream (options) {
+    // check if stream exists
+    if (typeof this.stream !== 'undefined') { return this.stream } else {
+      // default is get audio and video
+      if (!options) { options = { audio: true, video: true } }
+
+      // creating the stream and error-handling
+      try {
+        this.stream = await navigator.mediaDevices.getUserMedia(options)
+        return this.stream
+      } catch (err) {
+        console.log('Error capturing user media:',
+          err.name, err.message)
+      }
+    }
+  }
+
+  static async recordAudio (inputAudioId, outputAudioId, record) {
+    if (typeof this.mediaRecorder !== 'undefined') {
+      if (record) { this.mediaRecorder.start() } else { this.mediaRecorder.stop() }
+    } else {
+      const stream = await this.getStream({ audio: true })
+
+      // recorded audio referenced by 'audio' variable
+      const audio = document.getElementById(inputAudioId)
+
+      // Newer browsers
+      if ('srcObject' in audio) { audio.srcObject = stream }
+      // Older browsers
+      else { audio.src = window.URL.createObjectURL(stream) }
+
+      // Record audio given through stream
+      this.mediaRecorder = new MediaRecorder(stream)
+
+      this.mediaRecorder.start()
+
+      // Array to store the audio data
+      let audios = []
+
+      // Store new audios in array
+      this.mediaRecorder.ondataavailable = (newAudio) => {
+        audios.push(newAudio.data)
+      }
+
+      // Convert the audio data in the blob to mp3
+      // after stopping the recording
+      this.mediaRecorder.onstop = () => {
+        const savedAudio = document.getElementById(outputAudioId)
+
+        // blob of type mp3
+        const audioData = new Blob(audios,
+          { type: 'audio/mp3;' })
+
+        // clean up audios
+        audios = []
```

Comment:
don't think this is necessary. `audios` is a local variable (scoped to the `recordAudio` method) each time we run that method it will create a new audios array (so there's no need to clear it here, least not that i can tell)

Reply:
Actually I tested removing this line and it didn't empty the audio array. I think it's because the function creates one array which is being used throughout the next function calls. This makes sense, given that the recordAudio method checks if the mediaRecorder object exists before creating  a new one. The same thing must happen with the audios array then.